pkg/download/protocol.go

```diff
@@ -76,17 +85,31 @@ func (p *protocol) List(ctx context.Context, mod string) ([]string, error) {
 	var strList, goList []string
 	var sErr, goErr error
 	var wg sync.WaitGroup
-	wg.Add(2)
 
+	/*
+		TODO: potential refactor:
+
+		Storage Lister: just return stuff from storage, or error otherwise.
+
+		FallbackVCS lister: list from VCS, return empty list if error.
+
+		StrictVCS Lister: list from VCS, error if it doesn't succeed.
+
+		UnionLister(listers ...Lister): combines any number of listers.
+	*/
```

Comment:
@marwan-at-work did you decide to do this refactor? could maybe do this in a follow-up

Reply:
@arschles definitely planning on doing it 👍🏼

Reply:
ok. in this PR or later?